jenkins/docker/cpp.Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM symbolplatform/symbol-server-build-base:ubuntu-gcc-11-conan
+FROM symbolplatform/symbol-server-build-base:ubuntu-gcc-12-conan
```

Comment:
should we rename build-base to drop numbered versions too ? 
(can be separate PR ofc)

Reply:
Yea, I was thinking dropping the numbered version also.  The main issue would be that if the latest image is broken during an upgrade then CI will also be broken

Reply:
how is that different than if base image is broken now?

Reply:
Currently CI will use a specific base image plus conan packages.  If new compiler(new base image) or packages is been tested on a branch it will not affect the other branches.
If the versioned number is removed then when compiler gets updated, all jobs will be using the same base image.
Its only compiler updates so not a big deal.   I can remove update it in another pr.